File: Source/Client/Forms/frmEditor_Moral.vb
Imports System.Windows.Forms
Imports Core

Public Class frmEditor_Moral
#Region "Form Code"

    Private Sub BtnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        MoralEditorOk()
        Dispose()
    End Sub

    Private Sub BtnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        MoralEditorCancel()
        Dispose()
    End Sub

    Private Sub frmEditor_Moral_FormClosing(sender As Object, e As FormClosingEventArgs) Handles MyBase.FormClosing
        MoralEditorCancel()
    End Sub

    Private Sub BtnDelete_Click(sender As Object, e As EventArgs) Handles btnDelete.Click
        Dim tmpindex As Integer

        ClearMoral(GameState.EditorIndex)

        tmpindex = lstIndex.SelectedIndex
        lstIndex.Items.RemoveAt(GameState.EditorIndex - 1)
        lstIndex.Items.Insert(GameState.EditorIndex - 1, GameState.EditorIndex & ": " & Type.Moral(GameState.EditorIndex).Name)
        lstIndex.SelectedIndex = tmpindex

        MoralEditorInit()
    End Sub

    Private Sub lstIndex_Click(sender As Object, e As EventArgs) Handles lstIndex.Click
        MoralEditorInit()
    End Sub

    Private Sub frmEditor_Moral_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        lstIndex.Items.Clear()

        For i = 1 To MAX_MORALS
            lstIndex.Items.Add(i & ": " & Type.Moral(i).Name)
        Next
    End Sub

    Private Sub txtName_TextChanged(sender As Object, e As EventArgs) Handles txtName.TextChanged
        Dim tmpindex As Integer

        tmpindex = lstIndex.SelectedIndex
        Type.Moral(GameState.EditorIndex).Name = Trim$(txtName.Text)
        lstIndex.Items.RemoveAt(GameState.EditorIndex - 1)
        lstIndex.Items.Insert(GameState.EditorIndex - 1, GameState.EditorIndex & ": " & Type.Moral(GameState.EditorIndex).Name)
        lstIndex.SelectedIndex = tmpindex
    End Sub

    Private Sub chkCanCast_CheckedChanged(sender As Object, e As EventArgs) Handles chkCanCast.CheckedChanged
        Type.Moral(GameState.EditorIndex).CanCast = chkCanCast.Checked
    End Sub

    Private Sub chkCanPK_CheckedChanged(sender As Object, e As EventArgs) Handles chkCanPK.CheckedChanged
        Type.Moral(GameState.EditorIndex).CanPK = chkCanPK.Checked
    End Sub

    Private Sub chkCanPickupItem_CheckedChanged(sender As Object, e As EventArgs) Handles chkCanPickupItem.CheckedChanged
        Type.Moral(GameState.EditorIndex).CanPickupItem = chkCanPickupItem.Checked
    End Sub

    Private Sub chkCanDropItem_CheckedChanged(sender As Object, e As EventArgs) Handles chkCanDropItem.CheckedChanged
        Type.Moral(GameState.EditorIndex).CanDropItem = chkCanDropItem.Checked
    End Sub

    Private Sub chkCanUseItem_CheckedChanged(sender As Object, e As EventArgs) Handles chkCanUseItem.CheckedChanged
        Type.Moral(GameState.EditorIndex).CanUseItem = chkCanUseItem.Checked
    End Sub

    Private Sub chkDropItems_CheckedChanged(sender As Object, e As EventArgs) Handles chkDropItems.CheckedChanged
        Type.Moral(GameState.EditorIndex).DropItems = chkDropItems.Checked
    End Sub

    Private Sub chkLoseExp_CheckedChanged(sender As Object, e As EventArgs) Handles chkLoseExp.CheckedChanged
        Type.Moral(GameState.EditorIndex).LoseExp = chkLoseExp.Checked
    End Sub

    Private Sub chkPlayerBlock_CheckedChanged(sender As Object, e As EventArgs) Handles chkPlayerBlock.CheckedChanged
        Type.Moral(GameState.EditorIndex).PlayerBlock = chkPlayerBlock.Checked
    End Sub

    Private Sub chkNPCBlock_CheckedChanged(sender As Object, e As EventArgs) Handles chkNPCBlock.CheckedChanged
        Type.Moral(GameState.EditorIndex).NPCBlock = chkNPCBlock.Checked
    End Sub

    Private Sub cmbColor_SelectedIndexChanged(sender As Object, e As EventArgs) Handles cmbColor.SelectedIndexChanged
        Type.Moral(GameState.EditorIndex).Color = cmbColor.SelectedIndex
    End Sub

#End Region

End Class

File: Source/Client/Forms/frmEditor_Moral.cs
using System;
using System.Windows.Forms;
using Core;
using Microsoft.VisualBasic;

namespace Client
{

    public partial class frmEditor_Moral
    {
        public frmEditor_Moral()
        {
            InitializeComponent();
        }
        #region Frm

        protected override void WndProc(ref Message m)
        {
            const int WM_MOUSEACTIVATE = 0x0021;
            const int WM_NCHITTEST = 0x0084;

            if (m.Msg == WM_MOUSEACTIVATE)
            {
                // Immediately activate and process the click.
                m.Result = new IntPtr(1); // MA_ACTIVATE
                return;
            }
            else if (m.Msg == WM_NCHITTEST)
            {
                // Let the window know the mouse is in client area.
                m.Result = new IntPtr(1); // HTCLIENT
                return;
            }

            base.WndProc(ref m);
        }

        private void BtnSave_Click(object sender, EventArgs e)
        {
            Editors.MoralEditorOK();
            Dispose();
        }

        private void BtnCancel_Click(object sender, EventArgs e)
        {
            Editors.MoralEditorCancel();
            Dispose();
        }

        private void frmEditor_Moral_FormClosing(object sender, FormClosingEventArgs e)
        {
            Editors.MoralEditorCancel();
        }

        private void BtnDelete_Click(object sender, EventArgs e)
        {
            int tmpindex;

            Moral.ClearMoral(GameState.EditorIndex);

            tmpindex = lstIndex.SelectedIndex;
            lstIndex.Items.RemoveAt(GameState.EditorIndex);
            lstIndex.Items.Insert(GameState.EditorIndex, GameState.EditorIndex + 1 + ": " + Data.Moral[GameState.EditorIndex].Name);
            lstIndex.SelectedIndex = tmpindex;

            Editors.MoralEditorInit();
        }

        private void lstIndex_Click(object sender, EventArgs e)
        {
            Editors.MoralEditorInit();
        }

        private void frmEditor_Moral_Load(object sender, EventArgs e)
        {
            lstIndex.Items.Clear();

            for (int i = 0; i < Constant.MAX_MORALS; i++)
                lstIndex.Items.Add(i + 1 + ": " + Data.Moral[i].Name);
        }

        private void txtName_TextChanged(object sender, EventArgs e)
        {
            int tmpindex;

            tmpindex = lstIndex.SelectedIndex;
            Data.Moral[GameState.EditorIndex].Name = Strings.Trim(txtName.Text);
            lstIndex.Items.RemoveAt(GameState.EditorIndex);
            lstIndex.Items.Insert(GameState.EditorIndex, GameState.EditorIndex + 1 + ": " + Data.Moral[GameState.EditorIndex].Name);
            lstIndex.SelectedIndex = tmpindex;
        }

        private void chkCanCast_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].CanCast = chkCanCast.Checked;
        }

        private void chkCanPK_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].CanPK = chkCanPK.Checked;
        }

        private void chkCanPickupItem_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].CanPickupItem = chkCanPickupItem.Checked;
        }

        private void chkCanDropItem_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].CanDropItem = chkCanDropItem.Checked;
        }

        private void chkCanUseItem_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].CanUseItem = chkCanUseItem.Checked;
        }

        private void chkDropItems_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].DropItems = chkDropItems.Checked;
        }

        private void chkLoseExp_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].LoseExp = chkLoseExp.Checked;
        }

        private void chkPlayerBlock_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].PlayerBlock = chkPlayerBlock.Checked;
        }

        private void chkNpcBlock_CheckedChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].NpcBlock = chkNpcBlock.Checked;
        }

        private void cmbColor_SelectedIndexChanged(object sender, EventArgs e)
        {
            Data.Moral[GameState.EditorIndex].Color = (byte)cmbColor.SelectedIndex;
        }

        #endregion

    }
}